addons/mail/static/src/core/common/store_service.js

```diff
@@ -113,15 +113,20 @@ export class Store extends BaseStore {
 
     updateBusSubscription() {
         const channelIds = [];
+        let serverUpdated = false;
         const ids = Object.keys(this.Thread.records).sort(); // Ensure channels processed in same order.
         for (const id of ids) {
             const thread = this.Thread.records[id];
-            if (thread.model === "discuss.channel" && thread.hasSelfAsMember) {
+            if (thread.model === "discuss.channel") {
                 channelIds.push(id);
+                if (!thread.hasSelfAsMember) {
+                    this.env.services["bus_service"].addChannel(id);
```

Comment:
I tried to debounce the `_updateChannel` function which is called by `addChannel` because it is in a loop and we'd like to prevent spamming the server. However it did not work and bus tests kept failing because either it was invoked too late or with an outdated context. Another solution would be to refactor `addChannel` and turn it into `addChannels` that would aggregate channels before calling `_updateChannel`.

Reply:
@tsm-odoo what do you think?

Reply:
I still think that debouncing `_updateChannels` is the best option. I don't want to bloat the bus API which is already too broad. Moreover I think addChannel(s) methods are redundant and quite confusing. I tried using `debounce` and I figured adding a single `nextTick` before resolving the `waitForChannels` deferred works as expected.

The issue with those tests is that they are waiting for the channel to be added but not for the subscription to reach the server). Ideally, we should wait for the subscription to be done (calling `waitForSubscribe` method before resolving) and add an optional parameter to allow bypassing this behavior (some tests are waiting for a channel to be added, but no subscription is done since it was already done).

I guess the `nextTick` is good enough for now though.